RetailCoder.VBE/UnitTesting/AssertHandler.cs

```diff
@@ -10,41 +11,33 @@ public static class AssertHandler
 
         public static void OnAssertSucceeded()
         {
-            var handler = OnAssertCompleted;
-            if (handler != null)
-            {
-                handler(null, new AssertCompletedEventArgs(TestOutcome.Succeeded));
-            }
+            OnAssertCompleted?.Invoke(null, new AssertCompletedEventArgs(TestOutcome.Succeeded));
         }
 
         public static void OnAssertFailed(string message, [CallerMemberName] string methodName = "")
         {
-            var handler = OnAssertCompleted;
-            if (handler != null)
-            {
-                handler(null,
+            OnAssertCompleted?.Invoke(null,
                     new AssertCompletedEventArgs(TestOutcome.Failed,
                         string.Format(RubberduckUI.Assert_FailedMessageFormat, methodName, message).Trim()));
-                                
-            }
         }
 
         public static void OnAssertInconclusive(string message)
         {
-            var handler = OnAssertCompleted;
-            if (handler != null)
-            {
-                handler(null, new AssertCompletedEventArgs(TestOutcome.Inconclusive, message));
-            }
+            OnAssertCompleted?.Invoke(null, new AssertCompletedEventArgs(TestOutcome.Inconclusive, message));
         }
 
         public static void OnAssertIgnored()
         {
-            var handler = OnAssertCompleted;
-            if (handler != null)
-            {
-                handler(null, new AssertCompletedEventArgs(TestOutcome.Ignored));
-            }
+            OnAssertCompleted?.Invoke(null, new AssertCompletedEventArgs(TestOutcome.Ignored));
+        }
+
+        [DllImport("vbe7.dll", SetLastError = true)]
+        [return: MarshalAs(UnmanagedType.IUnknown)]
+        private static extern object rtcErrObj();
+
+        public static void RaiseVbaError(int number, string source = "", string description = "", string helpfile = "", int helpcontext = 0)
```

Comment:
++ I don't know why we never thought of importing this before. I just want to point out that, more than ever, we need to stick to our guns about pre-2007 being unsupported.